src/Ryujinx.Ui.Common/Helper/ShortcutHelper.cs

```diff
@@ -69,7 +69,7 @@ private static void CreateShortcutMacos(string appFilePath, byte[] iconData, str
             using StreamWriter scriptFile = new(scriptPath);
 
             scriptFile.WriteLine("#!/bin/sh");
```

Comment:
```suggestion
            string launchScript = $"""
                                   #!/bin/sh
                                   launch_arch="$(uname -m)"
                                   if [ "$(sysctl -in sysctl.proc_translated)" = "1" ]
                                   then
                                     launch_arch="arm64"
                                   fi

                                   arch -$launch_arch {basePath} {GetArgsString(appFilePath)}
                                   """;
            scriptFile.Write(launchScript);
```

What an annoying macOS bug. This is ugly but avoids setting an environment variable, as an alternative.